src/platform/Ameba/LwIPCoreLock.cpp

```diff
@@ -0,0 +1,60 @@
+/*
```

Comment:
Would you mind sharing this file with ESP32 ?

Reply:
Hi @mspang, do you mean sharing the file to ESP or using the same file with ESP32?
Thanks!

Reply:
I mean using the same file since the functionality seems to be identical.

Reply:
Hi @mspang , thanks for the suggestion. We had internal discussion about this, as this file may need to be modified later, so it is better that we avoid using the same file here.
Thanks!

Reply:
> Hi @mspang , thanks for the suggestion. We had internal discussion about this, as this file may need to be modified later, so it is better that we avoid using the same file here.
> Thanks!

It would not stop you from modifying the file later.

Reply:
Hi @mspang, in this case, we can use the same file.
Thanks!